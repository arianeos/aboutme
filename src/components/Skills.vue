<template>
  <div class="text-center my-16">
    <h2 class="pink--text text--lighten-1 pb-12 text-h2">
      Skills
    </h2>
    <v-chip
      v-for="skill in skills"
      :key="skill"
      large
      class="ma-2"
      color="blue darken-1"
      text-color="white"
    >
      {{ skill }}
    </v-chip>
  </div>
</template>
<script lang="ts">
import Vue from "vue";
import { mapState, mapActions } from "vuex";

export default Vue.extend({
  name: "Skills",
  created() {
    this.fetchSkills().catch(error => {
      console.error("An error occurred when fetching skills", error);
    });
  },
  computed: {
    ...mapState("skill", ["skills"])
  },
  methods: {
    ...mapActions("skill", ["fetchSkills"])
  }
});
</script>
