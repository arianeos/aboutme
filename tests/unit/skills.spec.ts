import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex, { Store } from "vuex";
import Skills from "@/components/Skills.vue";
import Vuetify from "vuetify";
import Vue from "vue";

const localVue = createLocalVue();

Vue.use(Vuetify);
localVue.use(Vuex);

describe("Skills", () => {
  let actions: any;
  let state;
  let store: Store<unknown>;

  beforeEach(() => {
    state = {
      skills: ["Ruby on Rails", "Vuejs"]
    };

    actions = {
      fetchSkills: jest.fn()
    };

    store = new Vuex.Store({
      modules: {
        skill: {
          state,
          actions,
          namespaced: true
        }
      }
    });
  });

  it("should renders the component and matches snapshot", () => {
    const wrapper = shallowMount(Skills, { store, localVue });
    expect(wrapper.html()).toMatchSnapshot();
    expect(actions.fetchSkills).toHaveBeenCalled();
  });
});
