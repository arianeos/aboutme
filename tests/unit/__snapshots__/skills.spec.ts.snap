// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills should renders the component and matches snapshot 1`] = `
<div class="text-center my-16">
  <h2 class="pink--text text--lighten-1 pb-12 text-h2">
    Skills
  </h2>
  <v-chip-stub color="blue darken-1" large="true" activeclass="" tag="span" active="true" closeicon="$delete" filtericon="$complete" textcolor="white" class="ma-2">
    Ruby on Rails
  </v-chip-stub>
  <v-chip-stub color="blue darken-1" large="true" activeclass="" tag="span" active="true" closeicon="$delete" filtericon="$complete" textcolor="white" class="ma-2">
    Vuejs
  </v-chip-stub>
</div>
`;
